### ISSUE TITLE
[BUG] n_best_forecasters in ForecastingRandomizedSearchCV returns the same forecaster

### ISSUE THREAD

**Issue by kostasek** [sktime/sktime#4704]
**Describe the bug**

when running `ForecastingRandomizedSearchCV` together with `make_reduction` and set `return_n_best_forecasters=20`, for some reason when calling `n_best_forecaster`s_ it returns a list of tuples containing the same forecaster instead of the top 20 forecasters (having different hyperparameters). Is there something I am missing? 

**To Reproduce**

```python
params_nn = {
    'estimator__hidden_layer_sizes': [(10,),(10,10),(5,),(15,)],   
    'estimator__alpha': [0.0001, 0.001, 0.01],    
    'estimator__shuffle': [False] ,               
    'estimator__learning_rate': ['constant', 'invscaling', 'adaptive'],             
    'estimator__early_stopping': [True,False],     
    'estimator__max_iter': [ 500, 700,1000,2000],            
    'window_length' :np.arange(1,3),
    'estimator__solver' :['adam','lbfgs']}
nn = make_reduction(MLPRegressor(random_state=0,),strategy=strategy)
rand_nn = ForecastingRandomizedSearchCV(forecaster = nn,cv = splitter_HO,
                                         param_distributions=params_nn,scoring = MeanSquaredError(),n_jobs = -1,n_iter=n_iter,
                                       return_n_best_forecasters=20,verbose=2)
rand_nn.fit(y_train,fh=[1,2,3,4,5,6])

rand_nn.n_best_forecasters_
```
![sktimeissue](https://github.com/sktime/sktime/assets/78386537/7830dbc6-be32-4321-8c2c-7c688ba24f8a)

**Expected behavior**
It should output forecasters with different hyperparameters instead it returns forecasters with same hyperparameters, the numbering of the forecasters in the tuple is also the same

**Additional context**

**Versions**
<details>

Python dependencies:
          pip: 23.0.1
       sktime: 0.19.1
      sklearn: 1.2.2
        numpy: 1.24.3
        scipy: 1.10.1
       pandas: 1.5.3
   matplotlib: 3.7.1
       joblib: 1.2.0
  statsmodels: 0.13.5
        numba: 0.57.0
     pmdarima: 2.0.3
      tsfresh: 0.20.0
   tensorflow: 2.10.0
tensorflow_probability: None

</details>

**Comment by fkiraly** [comment]
Thanks for reporting!

What would be great is adding imports to this, so the code block executes when copy-pasted.
You could use onboard data from `sktime`, e.g., `from sktime.datasets import load_airline` etc.

**Comment by fkiraly** [comment]
Thanks!

**Comment by hazrulakmal** [comment]
It seems that the bugs only occur for the reduction estimator, as NaiveForecaster() works just fine. A shorter and faster script with the same problem is as follows:

```from sktime.datasets import load_airline
from sktime.forecasting.naive import NaiveForecaster
from sktime.forecasting.model_selection import ForecastingGridSearchCV, ExpandingWindowSplitter

y = load_airline()
fh = [1,2,3]
forecaster = NaiveForecaster()
param_grid = {"strategy" : ["last", "mean", "drift"]}
gscv = ForecastingGridSearchCV(
    forecaster=forecaster,
    param_grid=param_grid,
    cv= xpandingWindowSplitter(fh=fh),
    return_n_best_forecasters=3)
gscv.fit(y)
gscv.n_best_forecasters_
```
works properly as expected returning the following 
```
[('1', NaiveForecaster()),
 ('2', NaiveForecaster(strategy='drift')),
 ('3', NaiveForecaster(strategy='mean'))]
```

After some digging, I suspect that the problem does not lie in the logic of SearchCV, but rather in the `clone` method of the `make_reduction` estimator, as shown in the example script below.
```
original_model = make_reduction(GradientBoostingRegressor(random_state=5,learning_rate=0.02))
original_model.fit(y)
copy_model = original_model.clone().set_params(estimator__random_state=42, estimator__learning_rate=0.01)
original_model.get_params()
```
However will return a dictionary that shows random_state=42 and learning_rate=0.01, which suggests that the clone method is not properly "cloning" the estimator.

**Comment by fkiraly** [comment]
ah, I see - the problem is with the nesting logic of `clone`, as it would not properly clone `sklearn` estimators.

This is coming from `skbase`. For now, I will just revert to `sklearn`'s `clone` until we can find sth better.

**Comment by fkiraly** [comment]
Fixing here: https://github.com/sktime/skbase/pull/195

**Comment by fkiraly** [comment]
@hazrulakmal, I added your boiled down example as a test case in https://github.com/sktime/sktime/pull/4707 - it passes now with `scikit-base 0.4.6`, so I think it is fixed at the source now.

@kostasek, the random search should behave as intended if you upgrade `scikit-base` to 0.4.6

**Comment by kostasek** [comment]
@fkiraly, there still seems to be a problem with the numbers in the tuples, take a look.
![problem](https://github.com/sktime/sktime/assets/78386537/dd4a9335-e0fd-4d76-94e8-283c89360748)

**Comment by fkiraly** [comment]
well, that's odd, but at least the estimator issue got resolved 😐

**Comment by fkiraly** [comment]
hm, I looked into it, and it seems it is not a bug but potentially confusing user interface design.

What is happening internally is caused by the scores being exactly equal for some estimators, which is likely if there are "blind" parameters, i.e., parameters that do not influence model behaviour, and/or a small test set.

In that case, a somewhat confusing sequence of things happen:

* equally ranking estimators are put in buckets. in your example, the first six estimators occupy, at equal performance, ranks 1, 2, 3, 4, 5, 6.
* then, the arithmetic mean of ranks is assigned to all estimators in the bucket. That is, all the estimators get ranks 3.5, 3.5, ...
* then, this is rounded down and converted to string - resulting in the number "3"

I would not have expected this, instead I would have thought they appear with the max (not rounded down arithmetic average), i.e., they co-occupy the rank "1".

I think this is a bit confusing, but I'm not sure what the "should" state is here. What do you think should happen?

**Comment by kostasek** [comment]
I am not an expert on the issue, but for models having exactly equal performance the ranking can be based on the fit time. A warning can be issued notifying for that.

**Comment by fkiraly** [comment]
Makes sense, you suggest breaking the ties somehow. That still doesn't solve though what should happen if we do end up with ties? (maybe another user doesn't want to sort by fit time etc)

**Comment by kostasek** [comment]
Then, all estimators in a specific bucket can keep the minimum rank and then each one is assigned a letter eg `'1C' , '2B'`

**Comment by fkiraly** [comment]
I see, interesting idea, sth like that might work. Maybe not letters as they are hard to make fully general (if there are more than 26 etc), but sth like `'1-1'`, `1-2` would be general and follow your idea.

I think it would require deprecation though, people might be using the first index in code.
